tensorpack/utils/utils.py

```diff
@@ -116,3 +119,59 @@ def building_rtfd():
     """
     return os.environ.get('READTHEDOCS') == 'True' \
         or os.environ.get('TENSORPACK_DOC_BUILDING')
+
+
+def deprecated(text, eos=None):
+    """Log deprecation warning.
+    Args:
+        text (str, optional): information
+        eos (str, optional): end of service date as tuple "YYYY-MM-DD"
+
+    Example:
+        @deprecated("Explanation what to do instead.", "2017-11-4")
+        def foo(...):
+            pass
+
+        deprecated("deprecated_item")("This is an info about an alternative.", "2017-11-4")
+        deprecated("A sentence about a remark.", "2017-11-4")()
```

Comment:
Yep. But this does not cause the issue. My best guess is that `@functools.wraps(func)` is responsible for that. But I cannot test the python3 version.